src/App.tsx

```diff
@@ -1,12 +1,70 @@
 import React from 'react';
 import './App.scss';
 
-// import { getAll, get5First, getRed } from './api/goods';
-// or
-// import * as goodsAPI from './api/goods';
+import { getAll, get5First, getRedGoods } from './api/goods';
+import { Good } from './types/Good';
+import { GoodsList } from './Components/GoodsList/GoodsList';
 
-const App: React.FC = () => (
-  <h1>Dynamic list of Goods</h1>
-);
+type State = {
+  goods: Good[] | null,
+};
+
+class App extends React.Component<{}, State> {
+  state = {
+    goods: null,
+  };
+
+  getData = (func: () => Promise<Good[]>) => {
+    try {
+      func().then(result => this.setState({ goods: result }));
+    } catch {
+      throw Error('error');
+    }
```

Comment:
if you want to handle error, let's do it properly
you can see the examples here
https://mate-academy.github.io/fe-program/js/extra/fetch#throw-an-error-if-request-failed